lib/auto-languageclient.js

```diff
@@ -178,6 +178,16 @@ export default class AutoLanguageClient {
     return this.socket != null ? 'socket' : 'stdio';
   }
 
+  // Return the name of your root configuration key
+  getRootConfigurationKey(): string {
+    return '';
+  }
+
+  // Optionally filter/transform the configuration object before it is sent to the server
+  filterConfigurationObject(configuration: any): any {
```

Comment:
Should probably call this map or transform as it will likely be used to reshape?

Reply:
Yep, filter is the wrong word, i'll change it to "map"